include/ipc-client.h

```diff
@@ -24,6 +24,10 @@ char *get_socketpath(void);
  */
 int ipc_open_socket(const char *socket_path);
 /**
+ * Issues a single IPC command without reading response.
+ */
+void ipc_single_command_no_response(int socketfd, uint32_t type, const char *payload, uint32_t *len);
```

Comment:
We can probably avoid the deadlock without resorting to rate limiting, no? Is that really the problem?